test/e2e/nodepool_autorepair_test.go

```diff
@@ -0,0 +1,151 @@
+	return func(t *testing.T) {
+		t.Parallel()
+		g := NewWithT(t)
+		ctx, cancel := context.WithCancel(context.TODO())
+
+		originalNP := hyperv1.NodePool{}
+		defer func() {
+			t.Log("Context Cancelled in testNodePoolAutoRepair function")
+			cancel()
+		}()
+
+		// List NodePools (should exists only one)
+		nodePools := &hyperv1.NodePoolList{}
+		err := mgmtClient.List(ctx, nodePools, &crclient.ListOptions{
+			Namespace: guestCluster.Namespace,
+		})
+		g.Expect(err).NotTo(HaveOccurred(), "failed getting existant nodepools")
+		for _, nodePool := range nodePools.Items {
+			if !strings.Contains(nodePool.Name, "-test-") {
+				originalNP = nodePool
+			}
+		}
+		g.Expect(originalNP.Name).NotTo(ContainSubstring("test"))
+		awsNPInfo := originalNP.Spec.Platform.AWS
+
+		// Define a new Nodepool
+		nodePool := &hyperv1.NodePool{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "NodePool",
+				APIVersion: hyperv1.GroupVersion.String(),
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      guestCluster.Name + "-" + "test-autorepair",
+				Namespace: guestCluster.Namespace,
+			},
+			Spec: hyperv1.NodePoolSpec{
+				Management: hyperv1.NodePoolManagement{
+					UpgradeType: hyperv1.UpgradeTypeReplace,
+					AutoRepair:  true,
+				},
+				ClusterName: guestCluster.Name,
+				Replicas:    &oneReplicas,
+				Release: hyperv1.Release{
+					Image: guestCluster.Spec.Release.Image,
+				},
+				Platform: hyperv1.NodePoolPlatform{
+					Type: guestCluster.Spec.Platform.Type,
+					AWS:  awsNPInfo,
+				},
+			},
+		}
+
+		// Create NodePool for current test
+		err = mgmtClient.Create(ctx, nodePool)
+		if err != nil {
+			if !errors.IsAlreadyExists(err) {
+				t.Fatalf("failed to create nodePool %s with Autorepair function: %v", nodePool.Name, err)
+			}
+
+			// If the NodePool already exists...
+			// This should not happen in CI but it's useful in local tests
+			existantNodePool := &hyperv1.NodePool{}
+			err = mgmtClient.Get(ctx, crclient.ObjectKeyFromObject(nodePool), existantNodePool)
+			g.Expect(err).NotTo(HaveOccurred(), "failed getting existant nodepool")
+			err = mgmtClient.Delete(ctx, existantNodePool)
+			g.Expect(err).NotTo(HaveOccurred(), "failed to Delete the existant NodePool")
+			t.Logf("waiting for NodePools in-place update with NTO-generated MachineConfig")
+			err = wait.PollImmediateWithContext(ctx, 10*time.Second, 15*time.Minute, func(ctx context.Context) (bool, error) {
+				if ctx.Err() != nil {
+					return false, err
+				}
+				err = mgmtClient.Create(ctx, nodePool)
+				if errors.IsAlreadyExists(err) {
+					t.Logf("WARNING: NodePool still there, will retry")
+					return false, nil
+				}
+				return true, nil
+			})
+			t.Logf("Nodepool Recreated")
+			g.Expect(err).NotTo(HaveOccurred(), "failed to Create the NodePool")
+		}
+
+		numZones := int32(len(clusterOpts.AWSPlatform.Zones))
+		clusterOpts.NodePoolReplicas = 1
+		numNodes := clusterOpts.NodePoolReplicas * numZones
```

Comment:
how are the zones relevant for the NodePool you create?

Reply:
Actually is not relevant, let me fix it.